client/dashboard/domains/contact-details/contact-form.tsx

```diff
@@ -0,0 +1,342 @@
+	errors?: Partial< Record< keyof DomainContactDetails, string > >;
+	isSubmitting?: boolean;
+}
+
+export default function ContactForm( {
+	initialData,
+	onSubmit,
+	onCancel,
+	isSubmitting = false,
+}: ContactFormProps ) {
+	const { data: countryList } = useQuery( countryListQuery() );
+	const [ selectedCountryCode, setSelectedCountryCode ] = useState(
+		initialData?.countryCode ?? ''
+	);
+	const { data: statesList } = useQuery( statesListQuery( selectedCountryCode ) );
+
+	const [ formData, setFormData ] = useState< DomainContactDetails >(
+		initialData ?? { optOutTransferLock: false }
+	);
+
+	const formattedCountryList = countryList?.map( ( country ) => ( {
+		label: country.name,
+		value: country.code,
+	} ) );
+
+	const isDirty = ! isEqual( formData, initialData );
+
+	const handleSubmit = ( data: DomainContactDetails ) => {
+		onSubmit?.( data );
+	};
+
+	useEffect( () => {
+		if ( formData.countryCode ) {
+			setSelectedCountryCode( formData.countryCode as string );
+		}
+	}, [ formData.countryCode, setSelectedCountryCode ] );
+
+	// Memoize the custom Edit component for the state field to prevent focus loss
+	const StateFieldEdit = useCallback(
+		( { field, onChange, data, hideLabelFromVision }: any ) => {
+			const { id, getValue } = field;
+
+			if ( ! statesList || statesList?.length === 0 ) {
+				return (
+					<InputControl
+						__next40pxDefaultSize
+						label={ hideLabelFromVision ? '' : __( 'State' ) }
+						placeholder={ __( 'State' ) }
+						value={ getValue( { item: data } ) }
+						onChange={ ( value ) => onChange( { [ id ]: value } ) }
+					/>
+				);
+			}
+
+			// If the item data is not in the statesList, set the state to the first option
+			if ( ! statesList?.some( ( state ) => state.code === getValue( { item: data } ) ) ) {
+				onChange( { [ id ]: statesList[ 0 ]?.code } );
+			}
+
+			return (
+				<SelectControl
+					__next40pxDefaultSize
+					__nextHasNoMarginBottom
+					label={ hideLabelFromVision ? '' : __( 'State' ) }
+					value={ getValue( { item: data } ) }
+					options={
+						statesList.map( ( state ) => ( {
+							label: state.name,
+							value: state.code,
+						} ) ) ?? []
+					}
+					onChange={ ( value ) => onChange( { [ id ]: value } ) }
+				/>
+			);
+		},
+		[ statesList ]
+	);
+
+	const fields: Field< DomainContactDetails >[] = [
+		{
+			id: 'firstName',
+			label: __( 'First Name' ),
+			type: 'text',
+			isValid: {
+				required: true,
+			},
+		},
+		{
+			id: 'lastName',
+			label: __( 'Last Name' ),
+			type: 'text',
+			isValid: {
+				required: true,
+			},
+		},
+		{
+			id: 'organization',
+			label: __( 'Organization' ),
+			type: 'text',
+		},
+		{
+			id: 'email',
+			label: __( 'Email' ),
+			type: 'email',
+			isValid: {
+				required: true,
+			},
+		},
+		{
+			id: 'phone',
+			label: __( 'Phone' ),
+			type: 'text',
+			isValid: {
+				required: true,
+			},
+		},
+		{
+			id: 'countryCode',
+			label: __( 'Country' ),
+			type: 'select',
```

Comment:
```suggestion
			type: 'text',
```
I think type refers to the returned value. This should be probably `text` 🤔